src/renderer.js

```diff
@@ -336,6 +336,15 @@ Renderer.prototype = {
       var featureHash = geo.hashFeature(d.properties[self.idField], group.tilePoint)
       if (!self.geometries[featureHash]) self.geometries[featureHash] = []
       self.geometries[featureHash].push(this)
+      if (d.geometry) { // Marker geometries have 'coordinates', not 'geometry'
+        if (d.geometry.type === 'Polygon' || d.geometry.type === 'MultiPolygon') {
+          d.properties['mapnik::geometry_type'] = 3
```

Comment:
What is 3, 2, and 1? Instead of using "magic numbers" I'd declare some constants in this file. Something like:

```
MAPNIK_GEOMETRY_TYPES = {
  POLYGON: 3,
  LINE: 2,
  POINT: 1
}
```